File: src/main/scala/io/github/jlprat/akka/http/workshop/bookstore/routes/CatalogManagerRoutes.scala
package io.github.jlprat.akka.http.workshop.bookstore.routes

import akka.http.scaladsl.server.{Directives, Route}

/**
  * This trait holds all routes in charge of managing the catalog
  */
trait CatalogManagerRoutes extends Directives with JsonProtocol with Authentication {

  val catalogManagerRoutes: Route = ???
}

File: src/main/java/io/github/jlprat/akka/http/workshop/java/bookstore/routes/CatalogManagerRoutes.java
package io.github.jlprat.akka.http.workshop.java.bookstore.routes;

import akka.actor.ActorRef;
import akka.http.javadsl.server.AllDirectives;
import akka.http.javadsl.server.Route;

/**
 * Created by jlprat on 23.04.17.
 */
public class CatalogManagerRoutes extends AllDirectives {

    private ActorRef catalogActorRef;

    public CatalogManagerRoutes(ActorRef catalogActorRef) {
        this.catalogActorRef = catalogActorRef;
    }

    public Route catalogManagerRoutes() {
        return null;
    }
}
